### PR TITLE
[RFR] Allow to use custom routes

### PR DESCRIPTION
**Pull request by manuquentin**

### CODE REVIEW

File: app/Component/ViewActions.js [hunk 1]
```diff
@@ -39,7 +39,6 @@ class ViewActions extends React.Component {
 }
 
 ViewActions.propTypes = {
-    view: React.PropTypes.object.isRequired,
```
Comment (jeromemacias) [inline app/Component/ViewActions.js]: You should just remove `.isRequired`.

Comment (jeromemacias) [inline app/Component/ViewActions.js]: And you can add `ViewActions.defaultProps = { view: null };`

Comment (manuquentin) [inline app/Component/ViewActions.js]: Done

File: app/ReactAdmin.js [hunk 2]
```diff
@@ -27,14 +29,22 @@ let routes = (
...
+            }
+        };
     }
     componentDidUpdate() {
         // stop progress bar
         Pace.stop();
     }
 
-    componentDidMount() {
+    componentWillReceiveProps() {
```
Comment (jeromemacias) [inline app/ReactAdmin.js]: Can't you use `componentWillMount()` ?

**Pull request merged**